src/inmanta/deploy/tasks.py

```diff
@@ -72,13 +73,33 @@ def get_executor_resource_details(self, version: int, resource_intent: "state.Re
         )
 
     async def get_executor(
-        self, task_manager: "scheduler.TaskManager", agent: str, resource_type: ResourceType, version: int
+        self,
+        *,
+        task_manager: "scheduler.TaskManager",
+        agent_spec: tuple[str, Collection[ResourceType]],
```

Comment:
I'm slightly on the fence about this one. I made it a tuple because they're conceptually coupled, and it prevents the confusion of a resource type argument, as well as an argument for all resource types. But it's slightly unconventional.